client-lib/src/main/java/com/ibm/ws/repository/strategies/writeable/AddThenHideOldStrategy.java

```diff
@@ -58,106 +53,56 @@ public AddThenHideOldStrategy(State desiredStateIfMatchingFound, State desiredSt
+                        repo.getMatchingResources(FilterPredicate.areEqual(FilterableAttribute.VANITY_URL, vanityURL));
+
+        boolean found = false;
+        for (RepositoryResource resource : matchingResources) {
+            // Only hide visible, published resources, and don't hide the one that's just been created!
+            if (!isVisible(resource)) {
+                continue;
             }
-            if (!oldResourceDeleted) {
-                // Since resource object is the resource that is being replaced, we need to get a copy of the
-                // resource before it gets updated and we can then delete that copy (the 4th parameter
-                // passed to the AdddThenDeleteStrategy constructor). There is a chance that the resource will
-                // have been deleted since we setup the cache so do this in a try catch.
-                RepositoryResource resourceToDelete;
-                try {
-                    resourceToDelete = resource.getRepositoryConnection().getResource(resource.getId());
-                } catch (RepositoryException e) {
-                    // Refresh the cache
-                    synchronized (allResources) {
-                        refreshCache(resource.getRepositoryConnection(), newResource.getId());
-                        resource = allResources.get(vanityURL);
-                        if (!(resource instanceof WebDisplayable)) {
-                            // Nothing matches any more so exit
-                            return;
-                        }
-                        resourceToDelete = resource.getRepositoryConnection().getResource(resource.getId());
-                    }
-                }
-                ((WebDisplayable) resource).setWebDisplayPolicy(DisplayPolicy.HIDDEN);
-                resource.uploadToMassive(new AddThenDeleteStrategy(null, State.DRAFT, true, (RepositoryResourceImpl) resourceToDelete));
+            if (!State.PUBLISHED.equals(((RepositoryResourceWritable) resource).getState())) {
+                continue;
             }
+            if (newResource.getId().equals(resource.getId())) {
+                continue;
+            }
+
+            if (found) {
+                // A resource has already been hidden, so there must have been multiple
+                // visible resources to start with. This suggests the repository is in a bad state
+                throw new RepositoryResourceUpdateException("There is more than 1 resource with the vanity URL " + newResource.getVanityURL() +
+                                                            " in state visible, " + newResource.getName() + "(" + newResource.getId() + ") and " +
+                                                            resource.getName() + "(" + resource.getId() + ")", newResource.getId());
+            }
+
+            found = true;
+            ((WebDisplayable) resource).setWebDisplayPolicy(DisplayPolicy.HIDDEN);
```

Comment:
At this point, I don't think you've checked that `resource` is an instance of `WebDisplayable`

Note that `isVisible` returns `true` for non-`WebDisplayable` resources.